autowarefoundation/autoware.universe
# feat: add publisher for detailed stop reason

## Description


Modify some planning modules to publish detailed reason of vehicle stop.
The additional message includes stop reason, its detail, and state.

The purpose of this change is to enable external application to recognize the reason of vehicle stop at each stop point.

Modified modules are the following:
* scene modules in behavior velocity planner, obstacle
* obstacle stop planner 
* surround obstacle checker

## Related links


Modify launcher for remap in [this PR](https://github.com/tier4/autoware_launch/pull/298)

## Tests performed
1. Run planning simulator.
2. Put initial pose and goal.
3. Open another terminal and run the launcher : `ros2 launch default_ad_api default_ad_api.launch.py`
4. Run the command : `ros2 topic echo /api/get/motion_factors`

The topic should be published as follows:

> ---
> header:
>   stamp:
>     sec: 1654647537
>     nanosec: 500246668
>   frame_id: map
> motion_factors:
> - pose:
>     position:
>       x: 3717.7839452345706
>       y: 73723.42854108683
>       z: 0.7956681883687983
>     orientation:
>       x: 0.0
>       y: 0.0
>       z: 0.23976677505715965
>       w: 0.9708305174327234
>   distance: 5.599999904632568
>   reason: 9
>   status: 2
>   detail: ''
> - pose:
>     position:
>       x: 0.0
>       y: 0.0
>       z: 0.0
>     orientation:
>       x: 0.0
>       y: 0.0
>       z: 0.0
>       w: 1.0
>   distance: .inf
>   reason: 3
>   status: 1
>   detail: ''
> ---

 



## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
